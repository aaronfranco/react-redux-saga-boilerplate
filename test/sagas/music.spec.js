import { expectSaga } from 'redux-saga-test-plan';

import music, { getJane } from 'sagas/music';
import { ActionTypes } from 'constants/index';

jest.mock('modules/connect', () => ({
  request: () => ({ items: [] }),
}));

describe('music', () => {
  it('should have the expected watchers', done => {
    expectSaga(music)
      .run({ silenceTimeout: true })
      .then(saga => {
        expect(saga).toMatchSnapshot();
        done();
      });
  });

  it('should have the getJane saga', () =>
    expectSaga(getJane, { payload: {  } })
      .put({
        type: ActionTypes.GET_JANE_SUCCESS,
        payload: {
            items: [] 
        },
      })
      .run());
});
