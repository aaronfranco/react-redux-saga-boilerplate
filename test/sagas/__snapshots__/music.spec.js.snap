// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`music should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "GET_JANE_REQUEST",
          "@@redux-saga-test-plan/json/function/getJane",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "GET_JANE_REQUEST",
      },
    },
  ],
}
`;
